curl-helper.c

```diff
@@ -2033,6 +2033,13 @@ static void handle_HTTP_VERSION(Connection *conn, value option)
     case 4:
 #if defined(CURL_HTTP_VERSION_2TLS)
       version = CURL_HTTP_VERSION_2TLS;
+#endif
+      break;
+    case 5:
+#if defined(CURL_HTTP_VERSION_2_PRIOR_KNOWLEDGE)
+      version = CURL_HTTP_VERSION_2_PRIOR_KNOWLEDGE;
+#else
+      caml_invalid_argument("CURLOPT_HTTP_VERSION: unsupported version CURL_HTTP_VERSION_2_PRIOR_KNOWLEDGE");
```

Comment:
1. It is not programmers error but runtime incompatibility, ie not invalid_argument
2. Is there really a reason to raise exn instead of silently continuing. What will the calling code do?

Reply:
1. fair enough
2. I assumed it would be better to fail fast as http/2 prior knowledge is a complete different protocol than http/1.1 so that it's clear what the error is ? but i guess i can turn into noop like other cases

Reply:
I mean what is caller code supposed to do with exn? In majority (all) cases they will fallback to query with whatever protocol possible. And other cases already chose noop for such things, so I would prefer that. It is still possible to assert on libcurl version.

Reply:
Anyway this does not work. The value is a an enum variant and cannot be checked with cpp. I wonder how the code above even works.

Reply:
good point :thinking:

Reply:
it means `CURL_HTTP_VERSION_2TLS` never worked :/
and now this PR raises minimum libcurl to 7.49.0 - this is no go, will need some configure guards or smth